### PR TITLE
Warn on invalid JSON file

### PR DESCRIPTION
**Pull request by kbariotis**
Re: #1144, I tried to make a more general error dialog that can be used in other places as well. 

Just set the `errorMessage` in app states' root and the error dialog will show up.

*To test, try to open an invalid JSON file*

Let me know what you think.

### CODE REVIEW

**Comment by vjeux** [comment]
Do you mind adding a screenshot by any chance? :) Thanks!

**Comment by Fausto95** [comment]
Capture d’écran 2020-04-01 à 23 19 34
cc: @vjeux 

Related: This is modal isn't showing up on dnd.

File: src/locales/en.json [hunk 1]
```diff
@@ -74,6 +74,7 @@
   "alerts": {
     "clearReset": "This will clear the whole canvas. Are you sure?",
     "couldNotCreateShareableLink": "Couldn't create shareable link.",
+    "couldNotLoadInvalidJSON": "Couldn't load invalid json",
```
Comment (lipis) [inline src/locales/en.json:77]:
```suggestion
    "couldNotLoadInvalidFile": "Couldn't load invalid file",
```

User's don't know about JSON :)

**Comment by lipis** [comment]
Can we do it that it throws the error on any file that is wrong?! I tried with images and other formats and it doesn't do anything.

File: src/data/blob.ts [hunk 2]
```diff
@@ -10,7 +11,7 @@ export async function loadFromBlob(blob: any) {
     try {
       const data = JSON.parse(contents);
       if (data.type !== "excalidraw") {
-        throw new Error("Cannot load invalid json");
+        throw new Error(t("alerts.couldNotLoadInvalidJSON"));
```
Comment (lipis) [inline src/data/blob.ts:14]:
```suggestion
        throw new Error(t("alerts.couldNotLoadInvalidFileN"));
```

File: src/data/blob.ts [hunk 3]
```diff
@@ -39,7 +40,7 @@ export async function loadFromBlob(blob: any) {
   }
   const { elements, appState } = updateAppState(contents);
   if (!elements.length) {
-    return Promise.reject("Cannot load invalid json");
+    return Promise.reject(t("alerts.couldNotLoadInvalidJSON"));
```
Comment (lipis) [inline src/data/blob.ts:43]:
```suggestion
    return Promise.reject(t("alerts.couldNotLoadInvalidFile"));
```

File: src/components/ErrorDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    }
+  }, [onClose]);
+
+  return (
+    <>
+      {modalIsShown && (
+        <Dialog
+          maxWidth={800}
+          onCloseRequest={handleClose}
+          title={t("buttons.export")}
```
Comment (lipis) [inline src/components/ErrorDialog.tsx:29]: Should be error or something

File: src/components/ErrorDialog.tsx [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        >
+          <div className="ExportDialog">{message}</div>
```
Comment (lipis) [inline src/components/ErrorDialog.tsx:31]: we should rename the class name to something more general if used in multiple places

**Comment by kbariotis** [comment]
@lipis the Open dialog shouldn't allow you to open any other file than `.json`. In theory you shouldn't be able to open an image for example. https://github.com/excalidraw/excalidraw/blob/master/src/data/json.ts#L44

**Comment by lipis** [comment]
Try drag and drop :)

**Comment by Fausto95** [comment]
Yea, dnd needs some fix. I'll get to it when I get some free time. Otherwise feel free to fix it

**Comment by kbariotis** [comment]
@lipis it works like a charm actually. @Fausto95 fixed it yesterday already :D https://github.com/excalidraw/excalidraw/pull/1159/files#diff-c8a101915dc6871f1f3d7ad2208d8655R250

File: src/components/ErrorDialog.tsx [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    if (onClose) {
+      onClose();
+    }
+  }, [onClose]);
+
+  return (
+    <>
+      {modalIsShown && (
+        <Dialog
+          maxWidth={800}
```
Comment (lipis) [inline src/components/ErrorDialog.tsx:27]: Let's make it much smaller.

**Comment by lipis** [comment]
> @lipis it works like a charm actually. @Fausto95 fixed it yesterday already :D [#1159 (files)](https://github.com/excalidraw/excalidraw/pull/1159/files#diff-c8a101915dc6871f1f3d7ad2208d8655R250)

What do you mean?! Drag and drop an image for example.. It should throw an error.. no?

**Comment by kbariotis** [comment]
haha yeah you are right sorry, rushed to talk :D fixed now

**Comment by Fausto95** [comment]
There still the case that you can upload a `json` file and the error doesn't show up

**Comment by lipis** [comment]
> There still the case that you can upload an invalid `json` file and the error doesn't show up

and you are working on it?

**Comment by kbariotis** [comment]
I couldn't find any other issue, neither with load scene or dnd.

**Comment by Fausto95** [comment]
@kbariotis Just checked again and it working fine.
 
Just one issue here, we store the whole state of the app in the localStorage, so we're now also storing the `errorMessage` property which we should not. That makes the modal pop up after we refresh the page

**Comment by dwelle** [comment]
@Fausto95 yea -- needs to be added to `clearAppStateForLocalStorage()`

**Comment by kbariotis** [comment]
Great catch @Fausto95, thank you, done.

**Pull request merged**